docs/source/developers/reviewing.rst

```diff
@@ -0,0 +1,226 @@
+  argument against a slow and straightforward reference implementation.
+
+- A mitigating concern, however, is the overall cost of running the test
+  suite.  Continuous Integration (CI) runtimes can be painfully long and
+  we should be wary of increasing them too much.  Sometimes it is
+  worthwhile to fine-tune testing parameters to balance the usefulness
+  of tests against the cost of running them (especially where stress tests
+  are involved, since they tend to imply execution over large datasets).
+
+
+Social aspects
+==============
+
+* Reviewing is a communication between the contributor and the reviewer.
+  Avoid letting questions or comments unanswered for too long ("too long"
+  is of course very subjective, but two weeks can be a reasonable
+  heuristic).  If you cannot allocate time soon, do say it explicitly.
+  If you don't have the answer to a question, do say it explicitly.
+  Saying "I don't have time immediately but I will come back later,
+  feel free to ping if I seem to have forgotten" or "Sorry, I am out of depth
+  here" is always better than saying nothing and letting the interlocutor
+  wondering.
+
+* If you know someone who may help on a blocking issue and past experience
+  suggests they can be available for that, feel free to add them to the
+  discussion (for example by cc'ing their Github handle).
+
+* If the contributor has stopped giving feedback or updating their PR,
+  perhaps they're not interested anymore, but perhaps also they're stuck
+  on some issue and feel unable to push their contribution any further.
+  Don't hesitate to ask ("I see this PR hasn't seen any updates recently,
+  are you stuck on something? Do you need any help?").
+
+* If the contribution is genuinely desirable and the contributor is not making
+  any progress, it is also possible to take it up.  Out of politeness,
+  it is however better to ask the contributor first.
+
+* Some contributors are looking for a quick fix to a specific problem and
+  don't want to spend too much time on it.  Others on the contrary are eager
+  to learn and improve their contribution to make it conform to the
+  project's standards.  The latter kind of contributors are especially
+  valuable as they may become long-term contributors or even committers
+  to the project; it can therefore be a good strategy to prioritize
+  interactions with such contributors.
+
+* Some contributors may respond "I will fix it later, can we merge anyway?"
+  when a problem is pointed.  Unfortunately, whether the fix is really
+  contributed soon in later PR is difficult to predict or enforce.  If
+  the contributor has shown to be trustable, it may be acceptable to do
+  as suggested.  Otherwise, it is better to decline the suggestion.
+
+* If a PR is generally ready for merge apart from trivial or uncontroversial
+  concerns, the reviewer may decide to push changes themselves to the
+  PR instead of asking the contributor to make the changes.
```

Comment:
Do we want a note about that same "it's best to ask for permission" or possibly we could frame this as "please don't take offense if this happens, it's common and is not intended to be a slight"

Reply:
I think the proposed phrase is out of scope for this document, as it's intended for reviewers not submitters.

Reply:
No change needed but I just wanted to mention that I would probably suggest this document to newer submitter as well.  Much of the content here is helpful for understanding why the code is written the way it is.  It would also help with common questions new contributors might face (e.g. how much should I be testing my code?)